tools/mayan_sync_helper/mayan_sync_transformer.py

```diff
@@ -0,0 +1,220 @@
+
+from importlib.metadata import metadata
+import xlrd
+import json
+
+
+class MetadataType:
+    def __init__(self, name, label):
+        self.name = name
+        self.label = label
+
+    def toJson(self):
+        return {"label": self.label,
+                "name": self.name}
+
+
+class MetadataReference:
+    def __init__(self, name, required):
+        self.name = name
+        self.required = required
+
+    def toJson(self):
+        return {"name": self.name,
+                "required": self.required}
+
+
+class DocumentType:
+    # label = fields.Str()
+    # metadata_types = fields.List(fields.Nested(MetadataType))
+
+    def __init__(self, name, label, purpose, display_order, categories):
+        self.name = name
+        self.label = label
+        self.purpose = purpose
+        self.display_order = display_order
+        self.categories = categories
+        self.metadata_references = []
+
+    def add_metadata(self, metadata_reference):
+        self.metadata_references.append(metadata_reference)
+
+    def toJson(self):
+        return {
+            "categories": [value for value in self.categories],
+            "display_order": self.display_order,
+            "label": self.label,
+            "purpose": self.purpose,
+            "metadata_types": [value.toJson() for value in self.metadata_references],
+            "name": self.name
+        }
+
+
+class MayanRequest:
+    def __init__(self, document_types, metadata_types):
+        self.document_types = document_types
+        self.metadata_types = metadata_types
+        self.remove_lingering_document_types = True
+        self.remove_lingering_metadata_types = True
+
+    def toJson(self):
+        return {"document_types": [value.toJson() for value in self.document_types],
+                "metadata_types": [value.toJson() for value in self.metadata_types],
+                "remove_lingering_document_types": self.remove_lingering_document_types,
+                "remove_lingering_metadata_types": self.remove_lingering_metadata_types
+                }
+
+
+def load_document_definitions():
+    # Give the location of the file
+    doctype_location = ("xml_definitions/doctype_definitions.xlsx")
```

Comment:
nit: perhaps a common function to get a file, open the workbook, extract the sheet, and possibly index to the first valid row? that would cut down on some of this duplicated code.